tools/autograd/gen_python_functions.py

```diff
@@ -447,6 +447,26 @@ def append_actuals_formals(actual, formal):
 
         env['AutoNoGIL'] = 'AutoNoGIL no_gil;' if not declaration['with_gil'] else ''
 
+        # Use the simple_return_type (Tensor) rather than the fancy return type
+        # (Tensor &).  This is important because the dispatch functions take
+        # mutable arguments *by value*, not by reference.  If you then return
+        # a a reference to such an argument, you will now have a pointer to a
+        # dangling stack entry.  Not good.
+        #
+        # You want:
+        #
+        #   Tensor dispatch_selu_(Tensor self) { return at::selu_(self); }
+        #
+        # *not*
+        #
+        #   Tensor& dispatch_selu_(Tensor self) { return at::selu_(self); }
+        #
+        # (NB: We can't make dispatch_selu_ take Tensor&, because the enclosing
```

Comment:
I think the code you want here is already supported if you pass `unpack_args=True` to `parse_arg`.  Might be better just to do that in this case (we already do it if there is a tensoroptions for example).  But this is probably fine too.